client/src/components/Dashboard.tsx

```diff
@@ -1,14 +1,86 @@
-import React, { ReactElement } from 'react';
+import React, { ReactElement, useEffect, useState } from 'react';
 import Nav from './Nav';
+import axios from 'axios';
+import Container from '@mui/material/Container';
+interface User {
+  id: number;
+  name: string;
+  firstName: string;
+  lastName: string;
+  email: string;
+  linkedinId: string;
+  githubId: string;
+  sub: string;
+  username: string;
+  picture: string;
+}
+interface Post {
+  id: number;
+  author: string;
+  userId: number;
+  title: string;
+  body: string;
+}
+
+interface Blog {
+  id: number;
+  title: string;
+  body: string;
+  userId: number;
+}
 
 const Dashboard = (): ReactElement => {
+  const [user, setUser] = useState<User>({} as User);
+  const [posts, setPosts] = useState<Post[]>([]);
+  const [postsCount, setPostsCount] = useState<number>(0);
+  const [followersCount, setFollowersCount] = useState<number>(0);
+  const [blogs, setBlogs] = useState<Blog[]>([]);
+  const [username, setUsername] = useState<string>('');
+
+  const getUser = () => {
+    if (user) {
+      axios
+        .get('/api/users/loggedIn')
+        .then(({ data }) => {
+          setUser(data);
+          setPostsCount(data.postsCount);
+          setFollowersCount(data.followersCount);
+        })
+        .catch((error) => {
+          console.error('Failed to get user:', error);
+        });
+    }
+  };
+
+  const checkUsername = () => {
+    return user.username === null || user.username === ''
+      ? user.name
+      : user.username;
+  };
+
+  useEffect(() => {
+    setUsername(checkUsername());
+  }, [user]);
+
+  useEffect(() => {
+    getUser();
+  }, [user]);
```

Comment:
Why did you seperate these into 2 useEffects?

Reply:
I thought the error I was getting was due to that. After I resolved the error I forgot to put them back together.